src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetAddressHandler.cs

```diff
@@ -0,0 +1,102 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using SFA.DAS.AdminService.Web.Infrastructure;
+using SFA.DAS.AdminService.Web.ViewModels.Roatp.Gateway;
+using SFA.DAS.AssessorService.Application.Api.Client.Clients;
+using SFA.DAS.AssessorService.ApplyTypes.Roatp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AdminService.Web.Handlers.Gateway
+{
+    public class GetAddressHandler : IRequestHandler<GetAddressRequest, AddressCheckViewModel>
+    {
+        private readonly IRoatpApplicationApiClient _applyApiClient;
+        private readonly IQnaApiClient _qnaApiClient;
+        private readonly IRoatpApiClient _roatpApiClient;
+        private readonly ILogger<GetOrganisationStatusHandler> _logger;
+
+        public GetAddressHandler(IRoatpApplicationApiClient applyApiClient, IQnaApiClient qnaApiClient, IRoatpApiClient roatpApiClient, ILogger<GetOrganisationStatusHandler> logger)
+        {
+            _applyApiClient = applyApiClient;
+            _qnaApiClient = qnaApiClient;
+            _roatpApiClient = roatpApiClient;
+            _logger = logger;
+        }
+
+        public async Task<AddressCheckViewModel> Handle(GetAddressRequest request, CancellationToken cancellationToken)
+        {
+            var pageId = GatewayPageIds.AddressCheck;
+
+            var model = new AddressCheckViewModel { ApplicationId = request.ApplicationId, PageId = pageId };
+           
+            var currentRecord = await _applyApiClient.GetGatewayPageAnswer(request.ApplicationId, pageId);
+            var applicationDetails = await _applyApiClient.GetApplication(model.ApplicationId);
+
+            if (currentRecord?.GatewayPageData != null)
+            {
+                model = JsonConvert.DeserializeObject<AddressCheckViewModel>(currentRecord.GatewayPageData);
```

Comment:
Are we duplicating the QnA data in the gateway page data whenever a page is complete?